R/compare.R

```diff
@@ -176,6 +157,44 @@ order_proxy <- function(proxy, direction = "asc", na_value = "largest") {
   }
 }
 
+# Order proxy ------------------------------------------------------------------
+
+#' Ordering proxy
+#'
+#' Returns a proxy object for ordering. For vctrs_vctr objects, this determines
+#' the behavior of `order()` and `sort()` (via `xtfrm()`). It also powers
+#' `vec_order()` and `vec_sort()` for other vector types.
```

Comment:
I'd be tempted to document the comparison and order proxy together so that you can more clearly describe the difference (i.e. they're the same except for lists, which we special case in order to allow ordering of data frames containing list-columns).